core/domain/stats_jobs_one_off.py

```diff
@@ -507,6 +868,219 @@ def reduce(exp_id, stringified_values):
                             num_completions_v1),)
 
 
+class StatisticsAuditV2(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off statistics audit for the v2 stats.
+
+    Performs a brief audit of exploration stats values to ensure that they
+    maintain simple sanity checks like being non-negative and so on.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [stats_models.ExplorationStatsModel]
+
+    @classmethod
+    def require_non_negative(
+            cls, exp_id, exp_version, property_name, value, state_name=None):
+        """Ensures that all the statistical data is non-negative.
+
+        Args:
+            exp_id: str. ID of the exploration.
+            exp_version: str. Version of the exploration.
+            property_name: str. The name used as the property's key in the
+                value dict.
+            value: dict. Its structure is as follows:
+                {
+                    'num_starts_v2': int. # of times exploration was
+                        started.
+                    'num_completions_v2': int. # of times exploration was
+                        completed.
+                    'num_actual_starts_v2': int. # of times exploration was
+                        actually started.
+                    'state_stats_mapping': A dict containing the values of
+                        stats for the states of the exploration. It is
+                        formatted as follows:
+                        {
+                            state_name: {
+                                'total_answers_count_v2',
+                                'useful_feedback_count_v2',
+                                'total_hit_count_v2',
+                                'first_hit_count_v2',
+                                'num_times_solution_viewed_v2',
+                                'num_completions_v2'
+                            }
+                        }
+                }
+            state_name: str|None. The name of the state whose statistics should
+                be checked.
+
+        Yield:
+            str. "Negative count: exp_id:? version:? state:? ?:?",
+                where ? is the placeholder for exp_id, exp_version,
+                property_name, state_name and value.
```

Comment:
switch state_name and property_name in this docstring (probably for v1 job as well)

Reply:
Done.